lib/fluent/test/base.rb

```diff
@@ -54,6 +54,7 @@ def initialize(klass, &block)
         end
         @instance.router = Engine.root_agent.event_router
         @instance.log = TestLogger.new
+        Engine.root_agent.instance_variable_set(:@log, @instance.log)
```

Comment:
This addition fixes holding different log object issue between `@instance` and `root_agent`.
If fix this problem, we should change Engine setup sequence in the tests.